hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -576,27 +579,34 @@ public Option<DeleteRecord> deltaMerge(DeleteRecord deleteRecord, BufferedRecord
     }
 
     @Override
-    public MergeResult<T> finalMerge(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord) throws IOException {
+    public BufferedRecord<T> finalMerge(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord) throws IOException {
       if (olderRecord.isDelete() || newerRecord.isDelete()) {
         if (shouldKeepNewerRecord(olderRecord, newerRecord)) {
           // IMPORTANT:
           // this is needed when the fallback HoodieAvroRecordMerger got used, the merger would
           // return Option.empty when the new payload data is empty(a delete) and ignores its ordering value directly.
-          return new MergeResult<>(newerRecord.isDelete(), newerRecord.getRecord());
+          return newerRecord;
         } else {
-          return new MergeResult<>(olderRecord.isDelete(), olderRecord.getRecord());
+          return olderRecord;
         }
       }
       return mergeNonDeleteRecord(olderRecord, newerRecord);
     }
 
-    public abstract MergeResult<T> mergeNonDeleteRecord(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord) throws IOException;
+    public abstract BufferedRecord<T> mergeNonDeleteRecord(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord) throws IOException;
   }
 
   // -------------------------------------------------------------------------
   //  Utilities
   // -------------------------------------------------------------------------
 
+  // Special handling for SENTINEL record in Expression Payload
+  private static BufferedRecord SENTINEL = new BufferedRecord(null, null, null, null, false);
+
+  private static <T> BufferedRecord<T> getDeleteBufferedRecord(String recordKey, RecordContext<T> recordContext) {
```

Comment:
The `recordContext` looks like it is not used anymore, can it be removed from the method arguments?